src/backend/distributed/executor/multi_utility.c

```diff
@@ -781,20 +783,22 @@ PlanIndexStmt(IndexStmt *createIndexStatement, const char *createIndexCommand)
 
 		isDistributedRelation = IsDistributedTable(relationId);
 
-		/* ensure future lookups hit the same relation */
-		namespaceName = get_namespace_name(RelationGetNamespace(relation));
-		createIndexStatement->relation->schemaname = namespaceName;
-
-		heap_close(relation, NoLock);
```

Comment:
I don't think it's correct to remove this before the isDistributedRelation check - by the time the statement gets actually executed on the master, somebody could have created a distributed table somewhere in the search path.